lib/streamlit/caching/cache_utils.py

```diff
@@ -35,15 +51,52 @@
 )
 from .hashing import update_hash
 
+if TYPE_CHECKING:
+    from streamlit.delta_generator import DeltaGenerator
+
 _LOGGER = get_logger(__name__)
 
 
+@dataclass
+class ElementMsgData:
+    """A non-interactive element's message and related metadata for
```

Comment:
This is specifically for elements that are non-interactive or are you saying this msgData is not interactive?

Reply:
> non-interactive elements. I wish we had a term specifically for them

Maybe "static elements", or is it too general word for it?

Another suggestion is "smart elements" / "dumb elements" in analogy with "smart components" / "dumb components" in React

Reply:
"static elements" kinda works, but feels at odds with some of them being layout blocks that you actively change in the script